pkg/errors/types.go

```diff
@@ -0,0 +1,8 @@
+package errors
+
+import api_errors "k8s.io/apimachinery/pkg/api/errors"
+
+type KNError struct {
```

Comment:
using a generic error struct for now, each error could use its own struct if neededin the future